-------------------------------------------------------------------
Wed Nov 30 08:16:55 UTC 2022 - Daniel Garcia <daniel.garcia@suse.com>

- Disable test_dist_boxed, broken with latest version of pytest-xdist,
  gh#pytest-dev/pytest-cov#565

-------------------------------------------------------------------
Mon Nov 28 09:31:24 UTC 2022 - Daniel Garcia <daniel.garcia@suse.com>

- Disable failing tests (test_contexts) that fails because of the new version
  of python-coverage, gh#pytest-dev/pytest-cov#570

-------------------------------------------------------------------
Fri Nov 25 11:55:18 UTC 2022 - Daniel Garcia <daniel.garcia@suse.com>

- Remove support-coverage-62.patch
- Remove support-setuptools60.patch
- Update to 4.0.0:
  * --cov-fail-under no longer causes pytest --collect-only to fail Contributed
    by Zac Hatfield-Dodds in #511.
  * Dropped support for multiprocessing (mostly because issue 82408). This
    feature was mostly working but very broken in certain scenarios and made
    the test suite very flaky and slow.
  * There is builtin multiprocessing support in coverage and you can migrate to
    that. All you need is this in your .coveragerc:

       [run]
       concurrency = multiprocessing
       parallel = true
       sigterm = true

  * Fixed deprecation in setup.py by trying to import setuptools before
    distutils. Contributed by Ben Greiner in #545.
  * Removed undesirable new lines that were displayed while reporting was
    disabled. Contributed by Delgan in #540.
  * Documentation fixes. Contributed by Andre Brisco in #543 and Colin O'Dell
    in #525.
  * Added support for LCOV output format via --cov-report=lcov. Only works with
    coverage 6.3+. Contributed by Christian Fetzer in #536.
  * Modernized pytest hook implementation. Contributed by Bruno Oliveira in
    #549 and Ronny Pfannschmidt in #550.

-------------------------------------------------------------------
Sun Jul 17 20:14:59 UTC 2022 - Ben Greiner <code@bnavigator.de>

- Add support-setuptools60.patch gh#pytest-dev/pytest-cov#545

-------------------------------------------------------------------
Mon Dec 20 01:21:53 UTC 2021 - Steve Kowalik <steven.kowalik@suse.com>

- Add support-coverage-62.patch:
  * Support coverage >= 6.2 

-------------------------------------------------------------------
Wed Nov 10 13:05:15 UTC 2021 - Ondřej Súkup <mimi.vx@gmail.com>

- drop pytest-cov-pr453-coverage54.patch
- update to 3.0.0
 * drop support for py2 and python-3.5

-------------------------------------------------------------------
Tue Sep 14 08:45:47 UTC 2021 - Steve Kowalik <steven.kowalik@suse.com>

- Inject multibuild to avoid build cycles. 

-------------------------------------------------------------------
Thu Feb  4 09:58:52 UTC 2021 - Ben Greiner <code@bnavigator.de>

- Add pytest-cov-pr453-coverage54.patch to work with coverage 5.4
  gh#pytest-dev/pytest-cov#453

-------------------------------------------------------------------
Wed Jan 27 13:51:26 UTC 2021 - Markéta Machová <mmachova@suse.com>

- Update to 2.11.1
  * Fixed the missing coverage context when using subprocesses.
  * Bumped minimum coverage requirement to 5.2.1. This prevents reporting issues.
- Drop merged pytest62.patch

-------------------------------------------------------------------
Fri Jan 15 11:59:34 UTC 2021 - Markéta Machová <mmachova@suse.com>

- Add pytest62.patch to make testsuite compatible with new pytest

-------------------------------------------------------------------
Fri Aug 21 14:16:07 UTC 2020 - Marketa Calabkova <mcalabkova@suse.com>

- Update to 2.10.1
  * Support for ``pytest-xdist`` 2.0, which breaks compatibility with ``pytest-xdist`` before 1.22.3 (from 2017).
  * Fixed the ``LocalPath has no attribute startswith`` failure that occurred when using the ``pytester`` plugin
    in inline mode.
  * Dropped support for Python 3 < 3.4, Pytest < 3.5 and Coverage < 4.4.

-------------------------------------------------------------------
Mon Jun 29 09:21:38 UTC 2020 - Marketa Calabkova <mcalabkova@suse.com>

- Update to 2.10.0
  * Improved the --no-cov warning. Now it's only shown if --no-cov is present before --cov.
  * Removed legacy pytest support. Changed setup.py so that pytest>=4.6 is required.

-------------------------------------------------------------------
Tue Jun  9 08:07:16 UTC 2020 - Tomáš Chvátal <tchvatal@suse.com>

- Update to 2.9.0:
  * Fixed RemovedInPytest4Warning when using Pytest 3.10. Contributed by Michael Manganiello in #354.
  * Made pytest startup faster when plugin not active by lazy-importing. Contributed by Anders Hovmöller in #339.
  * Various CI improvements. Contributed by Daniel Hahler in #363 and #364.
  * Various Python support updates (drop EOL 3.4, test against 3.8 final). Contributed by Hugo van Kemenade in #336 and #367.
  * Changed --cov-append to always enable data_suffix (a coverage setting). Contributed by Harm Geerts in #387.
  * Changed --cov-append to handle loading previous data better (fixes various path aliasing issues).
  * Various other testing improvements, github issue templates, example updates.
  * Fixed internal failures that are caused by tests that change the current working directory by ensuring a consistent working directory when coverage is called. See #306 and coveragepy#881

-------------------------------------------------------------------
Wed Jan 15 12:50:24 UTC 2020 - Tomáš Chvátal <tchvatal@suse.com>

- Skip test_cover_looponfail as it fails with new pytest

-------------------------------------------------------------------
Thu Dec  5 04:06:23 UTC 2019 - Steve Kowalik <steven.kowalik@suse.com>

- Skip 6 test cases that are causing problems with Python 3.8 

-------------------------------------------------------------------
Wed Nov  6 13:25:11 CET 2019 - Matej Cepl <mcepl@suse.com>

- Switch off another test cases to make test suite pass.
  gh#pytest-dev/pytest-cov#358

-------------------------------------------------------------------
Mon Oct  7 08:34:43 UTC 2019 - Tomáš Chvátal <tchvatal@suse.com>

- Update to 2.8.1:
  * Fixed #348 - regression when only certain reports (html or xml) are used then --cov-fail-under always fails.
  * Added compatibility with future xdist release that deprecates some internals (match pytest-xdist master/worker terminology). Contributed by Thomas Grainger in #321
  * Added support for Coverage 5.0

-------------------------------------------------------------------
Mon May  6 11:04:02 UTC 2019 - Tomáš Chvátal <tchvatal@suse.com>

- Update to 2.7.1:
  * Fixed AttributeError: 'NoneType' object has no attribute 'configure_node' error when --no-cov is used. Contributed by Alexander Shadchin in #263.
  * Various testing and CI improvements. Contributed by Daniel Hahler in #255, #266, #272, #271 and #269.
  * Improved documentation regarding subprocess and multiprocessing. Contributed in #265.
  * Improved pytest_cov.embed.cleanup_on_sigterm to be reentrant (signal deliveries while signal handling is running won't break stuff).
  * Added pytest_cov.embed.cleanup_on_signal for customized cleanup.
  * Improved cleanup code and fixed various issues with leftover data files. All contributed in #265 or #262.
  * Improved examples. Now there are two examples for the common project layouts, complete with working coverage configuration. The examples have CI testing. Contributed in #267.
  * Improved help text for CLI options.

-------------------------------------------------------------------
Wed Feb 13 09:49:04 UTC 2019 - Tomáš Chvátal <tchvatal@suse.com>

- Update to 2.6.1:
  * Added support for Pytest 4.1. Contributed by Daniel Hahler and Семён Марьясин in #253 and #230.
  * Various test and docs fixes. Contributed by Daniel Hahler in #224 and #223.
  * Fixed the "Module already imported" issue (#211). Contributed by Daniel Hahler in #228.

-------------------------------------------------------------------
Fri Oct 26 13:25:08 UTC 2018 - Ondřej Súkup <mimi.vx@gmail.com>

- update to 2.6.0
- cleanup deps
 * Dropped support for Python < 3.4, Pytest < 3.5 and Coverage < 4.4
 * Fixed some documentation formatting.
 * Added an example with addopts in documentation.
 * Fixed TypeError: 'NoneType' object is not iterable in certain xdist configurations.
 * Added a no_cover marker and fixture.
 * Fixed broken no_cover check when running doctests.
 * Fixed various issues with path normalization in reports
    (when combining coverage data from parallel mode). 
 * Report generation failures don't raise exceptions anymore.
    A warning will be logged instead.

-------------------------------------------------------------------
Sun Sep 24 21:47:40 UTC 2017 - arun@gmx.de

- update to version 2.5.1:
  * Fixed xdist breakage (regression in 2.5.0). Fixes #157.
  * Allow setting custom data_file name in .coveragerc. Fixes
    #145. Contributed by Jannis Leidel & Ionel Cristian Mărieș in
    #156.

- changes from version 2.5.0:
  * Always show a summary when --cov-fail-under is used. Contributed
    by Francis Niu in PR#141.
  * Added --cov-branch option. Fixes #85.
  * Improve exception handling in subprocess setup. Fixes #144.
  * Fixed handling when --cov is used multiple times. Fixes #151.

-------------------------------------------------------------------
Wed Mar  1 16:38:13 UTC 2017 - aloisio@gmx.com

- Update to version 2.4.0:
  * Added a "disarm" option: --no-cov. It will disable coverage
    measurements. Contributed by Zoltan Kozma in PR#135.
    WARNING: Do not put this in your configuration files, it's meant
    to be an one-off for situations where you want to disable coverage
    from command line.
  * Fixed broken exception handling on .pth file. See #136.
- Converted to single-spec
- Switched source URL to pypi.io

-------------------------------------------------------------------
Thu Oct 22 13:53:36 UTC 2015 - toddrme2178@gmail.com

- Update to version 2.2.0
  * Added support for changing working directory in tests.
    Previously changing working directory would disable coverage
    measurements in suprocesses.
  * Fixed broken handling for --cov-report=annotate.
- Update to version 2.1.0
  * Added support for coverage 4.0b2.
  * Added the --cov-append command line options. 
    Contributed by Christian Ledermann in PR#80.
- Update to version 2.0.0
  * Added --cov-fail-under, akin to the new fail_under option in
    coverage-4.0 (automatically activated if there’s a 
    [report] fail_under = ... in .coveragerc).
  * Changed --cov-report=term to automatically upgrade to 
    --cov-report=term-missing if there’s 
    [run] show_missing = True in .coveragerc.
  * Changed --cov so it can be used with no path argument 
    (in wich case the source settings from .coveragerc will 
    be used instead).
  * Fixed .pth installation to work in all cases 
    (install, easy_install, wheels, develop etc).
  * Fixed .pth uninstallation to work for wheel installs.
  * Support for coverage 4.0.
  * Data file suffixing changed to use coverage’s 
    data_suffix=True option (instead of the custom suffixing).
  * Avoid warning about missing coverage data 
    (just like coverage.control.process_startup).
  * Fixed a race condition when running with xdist 
    (all the workers tried to combine the files). 
    It’s possible that this issue is not present in 
    pytest-cov 1.8.X.

-------------------------------------------------------------------
Tue May 12 11:12:31 UTC 2015 - benoit.monin@gmx.fr

- update to version 1.8.1:
  * no changelog available
- update project URL
- increase minimum version of python-cov-core and python-pytest
- set a minimum version for python-py

-------------------------------------------------------------------
Wed Jul 30 07:27:46 UTC 2014 - toddrme2178@gmail.com

- Update to 1.7.0
  * No upstream changelog

-------------------------------------------------------------------
Mon Aug 12 09:35:04 UTC 2013 - speilicke@suse.com

- Initial version

